### PR TITLE
people_cooc: add files-per-author information to .yml

### PR DESCRIPTION
**Pull request by bzz**
Adding a _files-per-author_ table to `people_cooc` section of YAML.
Authors ordered by a number of files each changed, files ordered by path lexicographically. 

 - [x] add `author_files`
 - [x] add `requirments.txt`
 - [x] `go fmt ./cmd/hercules/main.go`

To avoid reviewing re-formatting changes use https://github.com/src-d/hercules/pull/7/files?w=1

**Test Plan**:
Build, rut with `-couples` and verify correctness of final .yml format
```
time ./hercules -couples https://github.com/src-d/go-git > gogit_couples.yaml
grep -A 10 "author_files" gogit_couples.yaml
python -c 'import yaml,sys;yaml.safe_load(sys.stdin)' < gogit_couples.yaml
```

### CODE REVIEW

**Comment by bzz** [comment]
\cc @vmarkovtsev for review

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+cycler==0.10.0
+matplotlib==2.0.2
+numpy==1.13.1
+pandas==0.20.3
+pyparsing==2.2.0
+python-dateutil==2.6.1
+pytz==2017.2
+PyYAML==3.12
+scipy==0.19.1
+seaborn==0.8
```
Comment (vmarkovtsev) [inline requirements.txt]: Luckily, we no longer need seaborn with matplotlib 2.0

File: README.md [hunk 2]
```diff
@@ -185,5 +185,10 @@ in-memory storage may require much RAM, for example, the Linux kernel takes over
 for the Linux kernel in "couples" mode is 1.5 GB and takes more than an hour / 180GB RAM to be
 parsed. However, most of the repositories are parsed within a minute.
 
+2. to use matplotlib in macOS and avoid errors us
```
Comment (vmarkovtsev) [inline README.md]: We should move it to "Plotting" section - there is already a way described with `--backend`, this is an alternative

File: couples.go [hunk 3]
```diff
@@ -105,12 +105,13 @@ func (couples *Couples) Consume(deps map[string]interface{}) (map[string]interfa
...
 	peopleMatrix := make([]map[int]int64, couples.PeopleNumber)
-	fmt.Printf("people_files:\n")
+	peopleFiles := make([][]string, couples.PeopleNumber)
 	for i := range peopleMatrix {
 		peopleMatrix[i] = map[int]int64{}
-		fmt.Printf("  %d: ", i)
 		for file, commits := range couples.people[i] {
-			fmt.Printf("%s, ", file)
+			//could be normalized further, by replacing file with idx in fileSequence
+			//but the would trade the space for readabilyt of result
```
Comment (vmarkovtsev) [inline couples.go]: I guess a misprint: **this** would .. **readability**

File: couples.go [hunk 4]
```diff
@@ -105,12 +105,13 @@ func (couples *Couples) Consume(deps map[string]interface{}) (map[string]interfa
 
 func (couples *Couples) Finalize() interface{} {
 	peopleMatrix := make([]map[int]int64, couples.PeopleNumber)
-	fmt.Printf("people_files:\n")
+	peopleFiles := make([][]string, couples.PeopleNumber)
```
Comment (vmarkovtsev) [inline couples.go]: How about keeping track of the number of times a developer changed a file? `[]map[string]int` Just asking

File: README.md [hunk 5]
```diff
@@ -181,9 +181,23 @@ python3 labours.py [--text-size] [--relative]
 1. Currently, go-git's file system storage backend is considerably slower than the in-memory one,
 so you should clone repos instead of reading them from disk whenever possible. Please note that the
 in-memory storage may require much RAM, for example, the Linux kernel takes over 200GB in 2017.
-2. Parsing YAML in Python is slow when the number of internal objects is big. `hercules`' output
+1. Parsing YAML in Python is slow when the number of internal objects is big. `hercules`' output
```
Comment (vmarkovtsev) [inline README.md]: But it's really second - the first is "1. Currently..." Or is markdown smart enough?

Comment (bzz) [inline README.md]: Yes, it is able to render the ordered lists this way.

You can check out how it looks like https://github.com/bzz/hercules/blob/1a43f1bfd0f6c369e298e95522843b538b20d875/README.md#caveats

File: couples.go (hunk 4 above)
Comment (bzz) [inline couples.go]: It's a nice idea but that would complicate the printing code, which sorts by file name and number of files. 
May be we could add it later

**Comment by bzz** [comment]
Thanks for prompt feedback, @vmarkovtsev ! Addressed in bf7da5b

**Comment by bzz** [comment]
That was fast :shipit:
Thanks!

**Pull request merged**